reflex/app.py

```diff
@@ -1317,3 +1320,40 @@ async def on_ping(self, sid):
         """
         # Emit the test event.
         await self.emit(str(constants.SocketEvent.PING), "pong", to=sid)
+
+
+def asset(relative_filename: str, dir: str = ".") -> str:
+    """Add an asset to the app.
+    Place the file next to your including python file.
+    Copies the file to the app's external assets directory.
+
+    Example:
+    ```python
+    rx.script(src=rx.asset("my_custom_javascript.js"))
+    rx.image(src=rx.asset("test_image.png","subfolder"))
+    ```
+
+    Args:
+        relative_filename: The relative filename of the asset.
+        dir: The directory to place the asset in.
+
+    Returns:
+        The relative URL to the copied asset.
+
+    """
+    calling_file = inspect.stack()[1].filename
+    cwd = Path.cwd()
+
+    module = inspect.getmodule(inspect.stack()[1][0])
+    assert module is not None
+
+    caller_module_path = module.__name__.replace(".", "/")
+    dir = caller_module_path if dir == "." else caller_module_path + "/" + dir
+    assets = constants.Dirs.APP_ASSETS
+    external = constants.Dirs.EXTERNAL_APP_ASSETS
+    Path(Path(cwd) / assets / external / dir).mkdir(parents=True, exist_ok=True)
```

Comment:
Create a variable for `Path(Path(cwd) / assets / external / dir)` first and use it in the next lines. Also check if the source exists with pathlib's `.resolve()` method

Reply:
@Lendemor I hope you are fine with using symlinks here, since we are not linking directories. That seems to be the only difference. https://docs.python.org/3/library/pathlib.html#pathlib.Path.symlink_to

@benedikt-bartscher I just tested what happens when `src_file` is not present. It already complains `No such file or directory`. OK, or wrap it into a an `AssetNotPresent`?

Reply:
@abulvenz awesome, if there is already a `FileNotFound` exception, I think we are fine, no additional wrapping needed

Reply:
@abulvenz Sorry for delay in review, was on break for a few days.

 It's passing windows CI so it should be fine I think.

I'm not sure about adding it inside `app.py` however, so could you move it to it's own file? Maybe inside `reflex/experimental` so we'll get some time to test it in real situations before making it official.

We'll also need to add unit tests for it.

Reply:
Thank you @Lendemor, After v0.5.0 everybody earned a small break. I will move it to experimental and try to add a unit test.